### PR TITLE
Update Quest: SoulBadgeQuest

### PR DESCRIPTION
**Pull request by Rympex**
- Get Soul Badge
- Buy Pokeballs on SafariStop
- Added 4 Random zone exp on Route 15 (Less Detectable)
- Fixed isDoable Condition

### CODE REVIEW

File: Quests/Kanto/SoulBadgeQuest.lua [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		if game.inRectangle(89,14,91,18) then--Zone 4
+			return moveToGrass()
+		else
+			return moveToCell(90,16)
+		end
+	end
+end
+
+function SoulBadgeQuest:PokecenterFuchsia()
+	self:pokecenter("Fuchsia City")
```
Comment (ghost) [inline Quests/Kanto/SoulBadgeQuest.lua:116]: This line makes the bot use pokecenter in a loop.

**Comment by Rympex** [comment]
?? what line?
i tested it 3 times and no problems. are u sure?

**Comment by ghost** [comment]
Ermmm I'm still testing, I'm not quite sure, I meant the line at the end of my note, it gets stuck sometimes after very specific conditions, can't identify them yet, sometimes bot run in and out of pokecenter, other time it keeps healing and healing at nurse joey.

**Comment by Rympex** [comment]
maybe u have pokemon without atk move and the condition is pokemonusable is false. and it try again heal
moves like magnitude are without power

**Comment by ghost** [comment]
Oh I see, I did introduced an Abra with only teleport move, I thought there was not going to be an issue cause Training function swaps to another pokemon when opponent level is higher than yours. But I guess thats why it kept healing, I changed `PokecenterFuchsia()` function to `return moveToMap("Fuchsia City")` and it got out the pokecenter but inmediately came back to it, I guess that was all, no issues to report now. Still maybe need a fix for this case? like also having a Magikarp only with Splash?

**Comment by ghost** [comment]
Finally tested, and worked :+1:

**Comment by c0pper** [comment]
Error during the script execution: Quests/Quest.lua:(213,7-96): Sould Badge quest has no method for map: Pokecenter Lavender

From what i can tell looks like it didn't set the pokecenter in fuchsia so on blackout it goes all the way back to lavender and doesn't know what to do. If i restart it manually it works again though.

**Comment by Rympex** [comment]
it is not a Error, I can not put every possible situation, just restart the script, and it will pick the right quest
 :)

**Comment by c0pper** [comment]
I'm currently using the script you uploaded on megafileupload today, with all your pulls included.
I'm at fuchsia with only 79$ and the bot is looping between route 18 and fuchsia city. I suppose it's trying to raise money to enter the safari zone but without success.

**Pull request closed without merging**